tensorrt_llm/_torch/models/modeling_llama.py

```diff
@@ -52,20 +59,148 @@ def __init__(
         else:
             pos_embd_params = None
 
-        super().__init__(
-            hidden_size=config.hidden_size,
-            num_attention_heads=config.num_attention_heads,
-            num_key_value_heads=config.num_key_value_heads,
-            max_position_embeddings=config.max_position_embeddings,
-            bias=config.attention_bias,
-            pos_embd_params=pos_embd_params,
-            layer_idx=layer_idx,
-            dtype=config.torch_dtype,
-            config=model_config,
-            use_qk_norm=use_qk_norm and not is_nope_layer,
-            aux_stream=aux_stream,
+        super().__init__(hidden_size=config.hidden_size,
+                         num_attention_heads=config.num_attention_heads,
+                         num_key_value_heads=config.num_key_value_heads,
+                         max_position_embeddings=config.max_position_embeddings,
+                         bias=config.attention_bias,
+                         pos_embd_params=pos_embd_params,
+                         layer_idx=layer_idx,
+                         dtype=config.torch_dtype,
+                         config=model_config)
+
+        if self.use_rope and self.use_qk_norm:
+            # here we must disable rope fusion regardless of attn_backend
+            self.enable_rope_fusion = False
+            self.rotary_emb = RotaryEmbedding(
+                RopeParams.from_config(config),
+                head_dim=self.head_dim,
+                is_neox=False,
+            )
+
+        if self.use_qk_norm:
+            self.head_dim = config.hidden_size // config.num_attention_heads
+            self.qk_norm = L2Norm(hidden_size=self.head_dim,
+                                  dtype=config.torch_dtype)
+        else:
+            self.qk_norm = None
+
+        self.attn_temperature_tuning = attn_temperature_tuning
+        self.floor_scale = getattr(config, "floor_scale", 8192.0)
+        self.attn_scale = getattr(config, "attn_scale", 0.1)
+
+    def _attn_qkv(self, q, k, v, attn_metadata, attention_mask, mrope_config,
+                  all_reduce_params):
```

Comment:
Add typing informations to inputs

Reply:
Thanks. 
I added the typing info there.